### DISCUSSION TITLE
Visualize Topic -1 in Barchart

### DISCUSSION THREAD

**Discussion by j-jsmith** [MaartenGr/BERTopic#1097] [Q&A] [answered]
Hi, 

I'm using BERTopic for my master's thesis for the analysis of politicians' tweets during election periods. 

So far, I left the stopwords in the text when using BERTopic and utilized the count vectorizer model when tokenizing topics to handle stopwords appearing in topics (I followed this guide which recommended this approach: [https://python.plainenglish.io/topic-modeling-for-beginners-using-bertopic-and-python-aaf1b421afeb]). 

As a consequence, wouldn't that mean that Topic -1 no longer shows outliers (most often "stopwords")? 
How can I interpret the result of Topic -1? For my analysis, it seems to be a relevant topic (words related to elections). 
Most importantly, is it possible to visualize Topic -1 in the bar chart? 

<img width="442" alt="My results" src="https://user-images.githubusercontent.com/123753926/225339237-6d0b397d-7f77-41e3-a34b-e7e6a2405c85.png">

Also, how can I adjust the y-axis ticks on the barchart? 

<img width="923" alt="y-axis ticks" src="https://user-images.githubusercontent.com/123753926/225342322-024c9ebc-9038-49cd-aab4-0ff44a660581.png">

I've only just recently joined the Python-community, so bare with me! 
Thanks so much in advance and really appreciate your help.

**Comment by MaartenGr** [comment, marked as answer]
> I've only just recently joined the Python-community, so bare with me!

No problem! Let's first start at the beginning, which is how BERTopic works. There is a [page in the documentation](https://maartengr.github.io/BERTopic/algorithm/algorithm.html) dedicated to how it works which I highly recommend reading through as, hopefully, that will make BERTopic a bit more intuitive in its usage. 

> So far, I left the stopwords in the text when using BERTopic and utilized the count vectorizer model when tokenizing topics to handle stopwords appearing in topics (I followed this guide which recommended this approach: [https://python.plainenglish.io/topic-modeling-for-beginners-using-bertopic-and-python-aaf1b421afeb]).
> As a consequence, wouldn't that mean that Topic -1 no longer shows outliers (most often "stopwords")?

Yes and no, and the reason for that is that the topics do not directly stem from the words themselves but from clusters of documents. What you are doing with removing stopwords that way is simply making sure that a topic will not contain the words but it will have no influence whatsoever on the generation of the clusters themselves. In other words, if you have a topic with keywords ["sports", "hockey", "the", "I"], removing stopwords will simply remove the keywords "the" and "I" but it will not change the meaning of topic but merely its **representation**.

> How can I interpret the result of Topic -1? For my analysis, it seems to be a relevant topic (words related to elections).
Most importantly, is it possible to visualize Topic -1 in the bar chart?

Typically, you do not interpret the result of Topic -1. Topic -1 consists of outliers that cannot be contained in a single topic but there is often not relationship between all documents in the outlier topic. For example, take the following image:

![image](https://user-images.githubusercontent.com/25746895/225582904-d96565e9-abb2-4b3d-957f-b40baef54a0a.png)

What we can see here is that many documents in Topic -1 (the grey points) are spread throughout and have very little structure to them. In other words, you do not want to analyze Topic -1 as a whole since the whole really does not make sense in most cases. Instead, you would want to analyze sub-clusters in Topic -1 or perhaps reduce them with something like `.reduce_outliers`. 

> Most importantly, is it possible to visualize Topic -1 in the bar chart?

That is currently not possible since Topic -1 is actually not a topic at all. The name is rather misleading but since it is essentially a bunch of random documents thrown together, there is little meaning to analyzing that topic from a global perspective.

**Reply by j-jsmith** [reply]
Thank you so much for your very fast response! It is extremely helpful!
